File: tests/csharp/tests-csharp-xunit-allure3-playwright/Tests/Ui/HeaderActiveNavTests.cs
using Allure.Net.Commons;
using Allure.Net.Commons.Attributes;
using Tests;

namespace Tests.Ui;

[AllureLabel("layer", "ui")]
[AllureEpic("Header")]
[AllureFeature("Active nav")]
[AllureSeverity(SeverityLevel.normal)]
[AllureSuite("Header active nav")]
public sealed class HeaderActiveNavTests : TestBase
{
    [Trait("TestCategory", "ui")]
    [Trait("TestCategory", "smoke")]
    [Fact(DisplayName = "Login page marks Login as the active header nav")]
    public void LoginPageMarksActiveLogin()
    {
        LoginPage.OpenPage()
            .Header.ShouldHaveActiveNav("header-nav-login");
    }

    [Trait("TestCategory", "ui")]
    [Fact(DisplayName = "Register page marks Register as the active header nav")]
    public void RegisterPageMarksActiveRegister()
    {
        RegisterPage.OpenPage()
            .Header.ShouldHaveActiveNav("header-nav-register");
    }

    [Trait("TestCategory", "ui")]
    [Fact(DisplayName = "Home page marks Home as the active header nav")]
    public void HomePageMarksActiveHome()
    {
        HomePage.OpenPage()
            .Header.ShouldHaveActiveNav("header-nav-home");
    }

    [Trait("TestCategory", "ui")]
    [Fact(DisplayName = "In-form Register link syncs the active header nav")]
    public void InFormRegisterLinkSyncsActiveNav()
    {
        LoginPage.OpenPage()
            .Header.ShouldHaveActiveNav("header-nav-login");
        LoginPage.ClickRegisterLink()
            .ShouldBeOpen()
            .Header.ShouldHaveActiveNav("header-nav-register");
    }

    [Trait("TestCategory", "ui")]
    [Fact(DisplayName = "In-form Login link syncs the active header nav")]
    public void InFormLoginLinkSyncsActiveNav()
    {
        RegisterPage.OpenPage()
            .Header.ShouldHaveActiveNav("header-nav-register");
        RegisterPage.ClickLoginLink()
            .ShouldBeOpen()
            .Header.ShouldHaveActiveNav("header-nav-login");
    }

    [Trait("TestCategory", "ui")]
    [Fact(DisplayName = "Header nav Register opens register and marks it active")]
    public void HeaderNavRegisterOpensRegister()
    {
        LoginPage.OpenPage()
            .Header.ClickNav("header-nav-register");
        RegisterPage.ShouldBeOpen()
            .Header.ShouldHaveActiveNav("header-nav-register");
    }

    [Trait("TestCategory", "ui")]
    [Fact(DisplayName = "Header nav Login opens login and marks it active")]
    public void HeaderNavLoginOpensLogin()
    {
        RegisterPage.OpenPage()
            .Header.ClickNav("header-nav-login");
        LoginPage.ShouldBeOpen()
            .Header.ShouldHaveActiveNav("header-nav-login");
    }
}

File: tests/java/tests-java-junit5-api_request-playwright/src/test/java/tests/ui/HeaderActiveNavTests.java
package tests.ui;

import annotations.Layer;
import io.qameta.allure.Epic;
import io.qameta.allure.Feature;
import io.qameta.allure.Severity;
import io.qameta.allure.SeverityLevel;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;
import tests.TestBase;

@Layer("ui")
@Epic("Header")
@Feature("Active nav")
@Severity(SeverityLevel.NORMAL)
@DisplayName("Header active nav")
class HeaderActiveNavTests extends TestBase {

    @Test
    @Tag("ui")
    @Tag("smoke")
    @DisplayName("Login page marks Login as the active header nav")
    void loginPageMarksActiveLogin() {
        app.login.open();
        app.header.shouldHaveActiveNav("header-nav-login");
    }

    @Test
    @Tag("ui")
    @DisplayName("Register page marks Register as the active header nav")
    void registerPageMarksActiveRegister() {
        app.register.open();
        app.header.shouldHaveActiveNav("header-nav-register");
    }

    @Test
    @Tag("ui")
    @DisplayName("Home page marks Home as the active header nav")
    void homePageMarksActiveHome() {
        app.home.open();
        app.header.shouldHaveActiveNav("header-nav-home");
    }

    @Test
    @Tag("ui")
    @DisplayName("In-form Register link syncs the active header nav")
    void inFormRegisterLinkSyncsActiveNav() {
        app.login.open();
        app.header.shouldHaveActiveNav("header-nav-login");
        app.login.clickRegisterLink();
        app.register.shouldBeOpen();
        app.header.shouldHaveActiveNav("header-nav-register");
    }

    @Test
    @Tag("ui")
    @DisplayName("In-form Login link syncs the active header nav")
    void inFormLoginLinkSyncsActiveNav() {
        app.register.open();
        app.header.shouldHaveActiveNav("header-nav-register");
        app.register.clickLoginLink();
        app.login.shouldBeOpen();
        app.header.shouldHaveActiveNav("header-nav-login");
    }

    @Test
    @Tag("ui")
    @DisplayName("Header nav Register opens register and marks it active")
    void headerNavRegisterOpensRegister() {
        app.login.open();
        app.header.clickNav("header-nav-register");
        app.register.shouldBeOpen();
        app.header.shouldHaveActiveNav("header-nav-register");
    }

    @Test
    @Tag("ui")
    @DisplayName("Header nav Login opens login and marks it active")
    void headerNavLoginOpensLogin() {
        app.register.open();
        app.header.clickNav("header-nav-login");
        app.login.shouldBeOpen();
        app.header.shouldHaveActiveNav("header-nav-login");
    }
}
